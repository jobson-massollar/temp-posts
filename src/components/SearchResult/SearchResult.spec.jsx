import { render, screen } from '@testing-library/react'
import { SearchResult } from '.'

describe('<SearchResult />', () => {

    it('should render many found', () => {
        const props = { search: true, qty: 2 }
        render(<SearchResult {...props}/>)

        expect(screen.getByText(/2 posts/)).toBeInTheDocument()
    })

    it('should render one found', () => {
        const props = { search: true, qty: 1 }
        render(<SearchResult {...props}/>)

        expect(screen.getByText(/1 post/)).toBeInTheDocument()
    })

    it('should render none found', () => {
        const props = { search: true, qty: 0 }
        render(<SearchResult {...props}/>)

        expect(screen.getByText(/no post/i)).toBeInTheDocument()
    })

    it('should be empty 1', () => {
        const props = { search: false, qty: 2 }
        const { container } = render(<SearchResult {...props}/>)

        expect(container.childElementCount).toBe(0)
    })

    it('should be empty 2', () => {
        const props = { search: true, qty: -1 }
        const { container } = render(<SearchResult {...props}/>)

        expect(container.childElementCount).toBe(0)
    })

    it('should match snapshot', () => {
        const props = { search: true, qty: 2 }
        const { container } = render(<SearchResult {...props}/>)

        expect(container.firstChild).toMatchSnapshot()
    })
})
