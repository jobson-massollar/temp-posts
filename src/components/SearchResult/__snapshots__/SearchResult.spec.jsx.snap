// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> should match snapshot 1`] = `
<div
  class="search-result-container"
>
  <p>
    2
     posts were found
  </p>
</div>
`;
